gamemodes/terrortown/gamemode/client/cl_voice.lua

```diff
@@ -25,6 +25,48 @@ local MutedState
 
 g_VoicePanelList = nil
 
+local function VoiceTryEnable()
+	if not hook.Run("TTT2CanUseVoiceChat", LocalPlayer()) then return end
+
+	if not VOICE.IsSpeaking() and VOICE.CanSpeak() then
+		VOICE.IsTeam = false
+		RunConsoleCommand("+voicerecord")
+	end
+end
+
+local function VoiceTryDisable()
+	if VOICE.IsSpeaking() and not VOICE.IsTeam then
+		RunConsoleCommand("-voicerecord")
+	end
+end
+
+local function VoiceTeamTryEnable()
+	if not hook.Run("TTT2CanUseTeamVoiceChat", LocalPlayer()) then return end
+
+	local ply = LocalPlayer()
+
+	if not IsValid(ply) then return end
+
+	local plyrd = ply:GetSubRoleData()
+
+	if not VOICE.IsSpeaking() and VOICE.CanSpeak() and ply:IsActive() and not plyrd.unknownTeam and not plyrd.disabledTeamVoice then
+		VOICE.IsTeam = true
+		RunConsoleCommand("+voicerecord")
+	end
+end
+
+local function VoiceTeamTryDisable()
+	if VOICE.IsSpeaking() and VOICE.IsTeam then
+		RunConsoleCommand("-voicerecord")
+	end
+end
+
+-- register a binding for the general voicechat
+bind.Register("ttt2_voice", VoiceTryEnable, VoiceTryDisable, "TTT2 Bindings", "f1_bind_voice", input.GetKeyCode(input.LookupBinding("+voicerecord") or KEY_X))
```

Comment:
Really great workaround using the new binding system (client based) to trigger the internal key system (shared) to support predictions in shared events! I‘m scared that I didn‘t thought about this a year ago when I tried reworking this. Gj!